src/irr/core/CDoublyLinkedList.h

```diff
@@ -0,0 +1,83 @@
+#ifndef __DOUBLY_LINKED_LIST_H_INCLUDED__
+#define __DOUBLY_LINKED_LIST_H_INCLUDED__
+#include "irr/core/alloc/PoolAddressAllocator.h"
+#include "irr/core/Types.h"
+
+namespace irr {
+	namespace core {
+
+		template<typename Value>
+		class DoublyLinkedList
+		{
+			struct Snode
+			{
+				struct Snode* prev;
+				Value data;
+				struct Snode* next;
```

Comment:
correct layout is
```cpp
Value data;
iterator_type prev,next;
```

because `Value` could be a type with complex alignment requirements (i.e. SSE/AVX data structure aligned to 16 or 32 bytes or more) and it would create padding inbetween your members